src/providers/LocalVideoProvider/index.tsx

```diff
@@ -32,25 +33,50 @@ const LocalVideoProvider: React.FC = ({ children }) => {
       setIsVideoEnabled(true);
     }
 
+    const observer: AudioVideoObserver = {
+      videoAvailabilityDidChange: (availability) => {
+        if (!availability.canStartLocalVideo) {
+          setIsReachVideoLimit(true);
+        } else {
+          setIsReachVideoLimit(false);
+        }
+        console.log(
+          'video availability changed: canStartLocalVideo ',
+          availability.canStartLocalVideo
+        );
+      },
+    };
+    audioVideo.addObserver(observer);
+
     return () => {
       setIsVideoEnabled(false);
+      audioVideo.removeObserver(observer);
     };
   }, [audioVideo]);
 
+  useEffect(() => {
+    if (isReachVideoLimit) {
```

Comment:
Do we need this? Seem like you can just do it inside the observer itself

Reply:
Yes, we need to have it in a separate `useEffect()` hook. The value of the state would not immediately change after  set state function which means if we do in in the observer, the state won't change in the log.

Reply:
> Yes, we need to have it in a separate `useEffect()` hook. The value of the state would not immediately change after  set state function which means if we do in in the observer, the state won't change in the log.

I think it's a good practice to wrap observer related logic in a stands alone UseEffect Hook.